src/prefab.test.ts

```diff
@@ -117,6 +121,41 @@ describe('poll', () => {
     await wait(frequencyInMs * 2);
     expect(prefab.pollCount).toEqual(2);
   });
+
+  it('is reset on init', async () => {
+    jest.spyOn(globalThis, 'clearTimeout');
+
+    const data = {values: {}};
+    const frequencyInMs = 25;
+    fetchMock.mockResponse(JSON.stringify(data));
+
+    const config: InitParams = {
+      apiKey: '1234',
+      context: new Context({user: {device: 'desktop'}}),
+    };
+
+    await prefab.init(config);
+
+    if (!prefab.loader) {
+      throw new Error('Expected loader to be set');
+    }
```

Comment:
You may want to write these as expectations instead?
```suggestion
    expect(prefab.loader).not.toBeNull();
```

Reply:
A reasonable thing to suggest. But if I do that, I still have to guard against this being `undefined` or when I later call `prefab.loader.context` typescript complains `error TS18048: 'prefab.loader' is possibly 'undefined'.`

There are other ways to handle this, but IME this is the simplest way to handle it in test code.

If you have a succinct preferred approach, please do tell!

Reply:
Ahhhh, good ol' typescript doesn't pick up the guards from the expectations in jest yet without writing wrapper functions per this PR: https://github.com/microsoft/TypeScript/pull/32695

👍 keep on trucking.